storage/rocksdb/rdb_utils.cc

```diff
@@ -308,4 +308,77 @@ bool rdb_database_exists(const std::string& db_name)
   return true;
 }
 
+/*
+  Set the patterns string.  If there are invalid regex patterns they will
+  be stored in m_bad_patterns and the result will be false, otherwise the
+  result will be true.
+*/
+bool Regex_list_handler::set_patterns(const std::string& pattern_str)
+{
+  bool pattern_valid= true;
+
+  // Create a normalized version of the pattern string with all delimiters
+  // replaced by the '|' character
+  std::string norm_pattern= pattern_str;
+  std::replace(norm_pattern.begin(), norm_pattern.end(), m_delimiter, '|');
+
+  // Make sure no one else is accessing the list while we are changing it.
+  mysql_rwlock_wrlock(&m_rwlock);
+
+  // Clear out any old error information
+  m_bad_pattern_str.clear();
+
+  try
+  {
+    // Replace all delimiters with the '|' operator and create the regex
+    // Note that this means the delimiter can not be part of a regular
+    // expression.  This is currently not a problem as we are using the comma
+    // character as a delimiter and commas are not valid in table names.
+    const std::regex* pattern= new std::regex(norm_pattern);
```

Comment:
True, this may be a decent improvement, but I wasn't really looking into improving the implementation, just dragging it down from the upstream handler.h into the MyRocks codebase as-is. I'll push a tested change later.